addOns/reports/src/main/java/org/zaproxy/addon/reports/sarif/SarifBigContentShrinker.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2021 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.reports.sarif;
+
+import java.util.Arrays;
+
+/**
+ * Represents a shrinker mechanism for big content data. It is used to avoid too big SARIF reports.
+ * Usage is e.g. inside web response body.
+ */
+public class SarifBigContentShrinker {
+
+    private static final String SHRINK_MARKER = "[...]";
+
+    /**
+     * Shrink given byte array to max size if necessary
+     *
+     * @param bytes the byte array to shrink
+     * @param maxAllowedBytes amount of maximum allowed bytes
+     * @return array with maximum allowed bytes or <code>null</code> of given array was <code>null
+     *     </code>
```

Comment:
Shouldn't these use `{@code null}`?

Reply:
Throughout.

Reply:
I am not sure what here is meant. You marked javadoc and mentioned `{@code null}` - please clarify.

Reply:
It should use a javadoc tag not html

Reply:
I use `<code>` a lot in javadoc and was not aware about that this shall not be used in ZAP.
When looking inside the existing ZAP source code I found 195 occurrences of "<code>" (including my ones) inside the workspace.

So shall I change all of my `<code>` elements? And what about `<h3>` and others?

Reply:
I hadn’t looked around. I’ll defer to others. @thc202 do you have preference?

Reply:
Use `{@code }` less verbose, it's not about the HTML generally.

Reply:
Found 7 matches in my code - will replace all parts with the `{@code null}` variant.